### DISCUSSION TITLE
Using H1 to control the cross slide

### DISCUSSION THREAD

**Discussion by EdFleta** [kachurovskiy/nanoels#93] [Q&A]
Hi Maxim, which lines of the code should be modified so as to use H1 software with no spindle encoder connected ?

so when I press the ON/OFF button the stepper should run either at the speed of one the threading presets or stop.

thanks for your help

ps this is not of course a 2 axis CNC attempt but rather a simple way to make the cross slide more usable (and I have a H1 kit already made!)

**Comment by kachurovskiy** [comment]
Hi @EdFleta, it would help to see some photos of what you've built to understand what you want. Sounds like you want to have something like "grinding" mode where the axis is moving with speed independent from the spindle - NanoEls doesn't have this feature yet.

**Reply by EdFleta** [reply]
nothing fancy about it: I have just installed the crosslide upgrade as described in one of your videos (printed parts and Nema 17).

I could control it with a simple arduino sketch (again not synchronised at all with H2 on leadscrew). This could help facing. Using H1 would allow to still keep some of the nice features of H1 (position memories and speed presets). 

The difficulty is that in auto mode (when ON is pressed) the stepper runs only when the spindle rotates. I'd like it to run independently.

In this configuration the cross slide will move forward and backward using the LEFT and RIGHT buttons (OFF) and continuously at the speed of the preset threading speed when pressing ON.

**Reply by kachurovskiy** [reply]
One way of achieving this is by using a timer to simulate the encoder running at all times. 

Add this in the `setup()`:

```
  // See pages 108 - 112 of ATmega328P_Datasheet.pdf
  // CS11 means prescaler = 8. WGM12 means issuing
  // interrupt when TCNT1 == OCR1A and set TCNT1 to 0.
  TCCR1B = (1 << WGM12) + (1 << CS11);
  TCCR1A = 0;
  TCNT1 = 0;
  OCR1A = 1250; // increasing this causes timer to trigger less frequently
  TIMSK1 |= (1 << OCIE1A); // enable timer
```

And add a new function that is triggered on this timer:
```
ISR(TIMER1_COMPA_vect) {
  spinEnc();
}
```

If this results in display text getting messed up (words in wrong lines), add `noInterrupts();` before `lcd.clear();` and `interrupts()` after `display.display();`.
